### PR TITLE
fix: share link on post page and renaming functions

### PR DESCRIPTION
**Pull request by nensidosari**
## Changes
- Fixing tracked short link when copying link from posts from post page

### Describe what this PR does
- Based on the reported issue in [slack](https://dailydotdev.slack.com/archives/C01L0QXQJNB/p1710745201272019)

All the removals of onShare are intentional as they are not used, all the renaming is intentional and will continue in a follow up PR so that all shares that just copy the link will be called copy link.

## Events

Did you introduce any new tracking events?
Don't forget to update the [Analytics Taxonomy sheet](https://docs.google.com/spreadsheets/d/18Lv7zXges9QfVX5VYL1a-Hyl0e1sQ3sLr0OK8YZWKXI/edit#gid=0)

Log the new/changed events below:

| Type   | event_name  | value |
|--------|-------------|-------|
| Change/New | event name  | extra: { ... } |

### On those affected packages:
- [x] Have you done sanity checks in the webapp?
- [ ] Have you done sanity checks in the extension?
- [ ] Does this not break anything in companion?

### Did you test the modified components media queries?
- [x] MobileL (420px)
- [x] Tablet (656px)
- [x] Laptop (1020px)

#### Did you test on actual mobile devices?
- [ ] iOS (Chrome and Safari)
- [ ] Android

MI-235 #done

### CODE REVIEW

File: packages/shared/src/components/PostOptionsMenu.tsx [hunk 1]
```diff
@@ -73,7 +72,7 @@ const PortalMenu = dynamic(
   },
 );
 
-export interface PostOptionsMenuProps extends ShareBookmarkProps {
```
Comment (nensidosari) [inline packages/shared/src/components/PostOptionsMenu.tsx:76]: Removing extend as onShare wasn't being used here

File: packages/shared/src/components/post/BasePostContent.tsx [hunk 2]
```diff
@@ -27,7 +27,7 @@ export function BasePostContent({
   customNavigation,
 }: BasePostContentProps): ReactElement {
   const { id } = post ?? {};
-  const { onCloseShare, sharePost, onSharePost } = engagementProps;
+  const { onCloseShare, sharePost, onCopyPostLink } = engagementProps;
```
Comment (nensidosari) [inline packages/shared/src/components/post/BasePostContent.tsx:30]: Have made this renaming, will continue with all renaming in the next PR that was a follow up for cleaning up after coly link experiment. https://dailydotdev.atlassian.net/browse/MI-233

File: packages/shared/src/components/post/PostActions.tsx [hunk 3]
```diff
@@ -26,7 +26,7 @@ import {
 } from '../../hooks/useBookmarkPost';
 
 export interface ShareBookmarkProps {
-  onShare: (post: Post) => void;
+  onShare?: (post?: Post) => void;
```
Comment (nensidosari) [inline packages/shared/src/components/post/PostActions.tsx:29]: Made optional as it wasn't used in many places.

File: packages/shared/src/components/post/PostMenuOptions.tsx [hunk 4]
```diff
@@ -10,13 +10,12 @@ import AuthContext from '../../contexts/AuthContext';
...
 import { SimpleTooltip } from '../tooltips/SimpleTooltip';
 import PostOptionsMenu, { PostOptionsMenuProps } from '../PostOptionsMenu';
-import { ShareBookmarkProps } from './PostActions';
 import { PromptOptions, usePrompt } from '../../hooks/usePrompt';
 import { Origin } from '../../lib/analytics';
 import useContextMenu from '../../hooks/useContextMenu';
 import { Button, ButtonVariant } from '../buttons/Button';
 
-export interface PostMenuOptionssProps extends ShareBookmarkProps {
+export interface PostMenuOptionssProps {
```
Comment (nensidosari) [inline packages/shared/src/components/post/PostMenuOptions.tsx:18]: onShare not being used

File: packages/shared/src/components/ShareMobile.tsx [hunk 5]
```diff
@@ -16,23 +16,24 @@ import { useAnalyticsContext } from '../contexts/AnalyticsContext';
...
 
 export function ShareMobile({ post, share, link }: Props): ReactElement {
-  const [copying, copyLink] = useCopyPostLink(link);
+  const [copying] = useCopyPostLink(link);
+
   const { trackEvent } = useAnalyticsContext();
 
   const onShare = () => {
     trackEvent(postAnalyticsEvent(AnalyticsEvent.StartShareToSquad, post));
-    share(post);
```
Comment (rebelchris) [inline packages/shared/src/components/ShareMobile.tsx:28]: I added this for one edge-case that was sync before.
Not sure where exactly, but maybe we refactored that part already. (just making sure it doesn't break)

File: packages/shared/src/hooks/usePostContent.ts [hunk 6]
```diff
@@ -35,13 +40,29 @@ const usePostContent = ({
...
   const onPostClick = useOnPostClick({ origin });
+  const { commentsPermalink } = post;
+  const cid = ReferralCampaignKey.SharePost;
+  const { getShortUrl } = useGetShortUrl();
+  const { sharePost, closeSharePost } = useSharePost(origin);
+  const [, copyLink] = useCopyPostLink();
+  const trackShareEvent = (provider: ShareProvider) =>
+    trackEvent(
+      postAnalyticsEvent('share post', post, {
+        extra: { provider, origin: Origin.ShareBar },
```
Comment (rebelchris) [inline packages/shared/src/hooks/usePostContent.ts:51]: Is the origin ok here?
Not sure where we use this hook from, but might need to pass origin from top level somewhere?

**Review by rebelchris — APPROVED** [review]
Minor comments not blocking

File: packages/shared/src/components/ShareMobile.tsx (hunk 5 above)
Comment (nensidosari) [reply packages/shared/src/components/ShareMobile.tsx:28]: I think this was using a different share function before that required the post, now it's using the usePostContent share that doesn't need one. Checked for collections, squads and normal posts.

P.s this share/copy thing is also a mess 😭

File: packages/shared/src/hooks/usePostContent.ts (hunk 6 above)
Comment (nensidosari) [reply packages/shared/src/hooks/usePostContent.ts:51]: This might need adjustment, verifying now

Comment (nensidosari) [reply packages/shared/src/hooks/usePostContent.ts]: It was passed, I had forgotten to use the right one here

**Pull request merged**